python/lsst/ts/mtdome/mock_llc/mock_motion/base_llc_motion.py

```diff
@@ -132,25 +109,184 @@ def set_target_position_and_velocity(
             [min position, max position] or if abs(crawl_velocity) > max_speed.
 
         """
-        if not self._min_position <= end_position <= self._max_position:
+        if motion_state not in [LlcMotionState.MOVING, LlcMotionState.CRAWLING]:
+            raise ValueError("motion_speed should be MOVING or CRAWLING.")
+
+        if (
+            motion_state == LlcMotionState.MOVING
+            and not self._min_position <= end_position <= self._max_position
+        ):
             raise ValueError(
                 f"The target position {end_position} is outside of the "
                 f"range [{self._min_position, self._max_position}]"
             )
-        if math.fabs(crawl_velocity) > self._max_speed:
-            raise ValueError(
-                f"The target speed {math.fabs(crawl_velocity)} is larger than the "
-                f"max speed {self._max_speed}."
-            )
 
         self._commanded_motion_state = motion_state
         self._start_tai = start_tai
         self._end_position = end_position
-        self._crawl_velocity = crawl_velocity
         duration = self._get_duration()
         self._end_tai = self._start_tai + duration
         return duration
 
+    def _get_distance(self) -> float:
+        """Determines the smallest distance [rad] between the initial and
+        target positions assuming motion around a circle.
```

Comment:
It does also work for elevation since that uses angles as well. But it will not work for the aperture shutter since that is a linear motion and doesn't use angles so the conversion to degrees would mess up the result. I added a note that systems that do not use radians should override the method.